realm/src/main/java/io/realm/Realm.java

```diff
@@ -421,13 +430,23 @@ private static Realm create(File writableFolder, String filename, byte[] key) {
         }
     }
 
-    private static Realm createAndValidate(String absolutePath, byte[] key, boolean validateSchema, boolean autoRefresh) {
+    private static synchronized Realm createAndValidate(String absolutePath, byte[] key, boolean validateSchema,
```

Comment:
@cmelchior and I have been discussing this for a while, not just for this PR.
If we don't synchronize this method concurrent initialization might occur, which would be very bad of course